### PR TITLE
Add metric for jvm memory usage after gc

### LINKED ISSUE #2900: Track JVM memory usage after GC
In talking about JVM gc metrics, the java SIG has concluded that it would be valuable to track memory usage after garbage collection. Discussion available [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6362#issuecomment-1262848079).

The JVM provides access to determine the memory usage by different memory pools as of the most recent garbage collection event. By recording this, we can analyze it over time to see if there is:

 1. Evidence of memory leak. The post-gc usage would grow over time, since garbage collection would be unable to reclaim certain chunks of memory.
 2. Evidence that the memory pool size is inadequate. If the post-gc heap is high as a percentage of the total memory limit, you may way to allocate more memory to avoid overly frequent garbage collections.

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #2900.

Add new metric `process.runtime.jvm.memory.usage_after_last_gc`.

### CODE REVIEW

**Comment by jack-berg** [comment]
cc @open-telemetry/java-instrumentation-maintainers @open-telemetry/java-maintainers

**Review by carlosalberto — APPROVED** [review]

File: specification/metrics/semantic_conventions/runtime-environment-metrics.md [hunk 1]
```diff
@@ -61,26 +61,28 @@ consider, for example pthreads vs green thread implementations.
...
+|                                            |                                                                                    |         |                                           |                                                   |            | pool          | Name of pool [1]      |
+| process.runtime.jvm.memory.limit           | Measure of max obtainable memory                                                   | Bytes   | `By`                                      | UpDownCounter                                     | Int64      | type          | `"heap"`, `"nonheap"` |
+|                                            |                                                                                    |         |                                           |                                                   |            | pool          | Name of pool [1]      |
+| process.runtime.jvm.memory.usage_after_gc  | Measure of memory used after the most recent garbage collection event on this pool | Bytes   | `By`                                      | UpDownCounter                                     | Int64      | type          | `"heap"`, `"nonheap"` |
```
Comment (trask) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:74]: what do u think?
```suggestion
| process.runtime.jvm.memory.usage_after_last_gc  | Measure of memory used after the most recent garbage collection event on this pool | Bytes   | `By`                                      | UpDownCounter                                     | Int64      | type          | `"heap"`, `"nonheap"` |
```

Comment (trask) [inline specification/metrics/semantic_conventions/runtime-environment-metrics.md:74]:
```suggestion
| process.runtime.jvm.memory.usage_after_gc  | Measure of memory used, as measured after the most recent garbage collection event on this pool | Bytes   | `By`                                      | UpDownCounter                                     | Int64      | type          | `"heap"`, `"nonheap"` |
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ release.
 
 - Handle duplicate description comments during Prometheus conversion.
   ([#2890](https://github.com/open-telemetry/opentelemetry-specification/pull/2890))
+- Add `process.runtime.jvm.memory.usage_after_gc` metric to semantic conventions.
```
Comment (trask) [inline CHANGELOG.md:21]:
```suggestion
- Add `process.runtime.jvm.memory.usage_after_last_gc` metric to semantic conventions.
```

**Review by trask — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by breedx-splk — COMMENTED** [review]
Sorry to drag this out. Do you think we can get away with spelling it `process.runtime.jvm.memory.usage_after_gc` maybe (`last` seemed redundant/given)? What about adding an optional attribute for the gc id?

**Comment by trask** [comment]
> Sorry to drag this out. Do you think we can get away with spelling it `process.runtime.jvm.memory.usage_after_gc` maybe (`last` seemed redundant/given)? What about adding an optional attribute for the gc id?

(guilty) see https://github.com/open-telemetry/opentelemetry-specification/pull/2901#discussion_r1007240933

`memory_after_gc` seemed ambiguous to me (e.g. is it the sum after each gc during the time period?), but definitely open to revisiting

**Comment by breedx-splk** [comment]
> `memory_after_gc` seemed ambiguous to me (e.g. is it the sum after each gc during the time period?), but definitely open to revisiting

Hah! Ok, let's stay the course. 🙊

**Review by reyang — APPROVED** [review]

**Pull request merged**